Possibly show Price in Minor charter

perwestling opened the pull request:
Show price if minor has a value and no owner.

perwestling commented:
If the minor has attribute 'value' non-zero, do show Price in charter:
![image](https://user-images.githubusercontent.com/2631151/122657175-5f8c5f80-d161-11eb-822d-9bc6fe572948.png)

If discount is non-zero, present Price as value - discount.

If value is zero, or owner is set, do not change current view. So it will looks like this:
![image](https://user-images.githubusercontent.com/2631151/122657201-95314880-d161-11eb-8bc0-d9330f6600da.png)

perwestling commented:
Commit also contains some updates to use the new value and discount attributes. As far as I know it is so far only 1893 that uses these.

tobymao commented on assets/app/view/game/corporation.rb:
```diff
@@ -311,6 +313,11 @@ def render_tokens
       end
 
+      def render_minor_price
+        price_str = "Price: #{@game.format_currency(@corporation.value - @corporation.discount)}"
```
should this be min_bid?

perwestling replied:
Fixed

tobymao commented:
why do minors have value and discount? shouldn't they be companies then?

perwestling commented:
> why do minors have value and discount? shouldn't they be companies then?

@tobymao Minors is in between Company and Corporation.
They do operate - but does not have any shares - but they have an owner.
They cannot be sold. They can have abilities. etc

In latter titles (1893, and maybe 1824) I have moved towards buying the minor instead of buying a company which is connected to a minor and indirectly taken over it.  

So I introduced possibility to buy Minor during the auction. 
But the Minor did not show the value/cost. If you clicked on them a Buy button appear, but you cannot do this for all.

In all it makes for an easier implementation, to use Minors directly instead of Proxy privates.

tobymao commented:
but don't all other games like 1846 use company and then convert to minor?

perwestling commented:
Ok, I will throw the changes with minors out, by making a commit that uses privates instead.
